pkg/route/secretmanager/manager_test.go

```diff
@@ -0,0 +1,300 @@
+			expectErr:  1,
+		},
+		{
+			name: "error while removing SecretEventHandler",
+			register: []routeSecret{
+				{routeName: "route1", secretName: "secret1"},
+			},
+			unregister: []routeName{"route1"},
+			sm: fake.SecretMonitor{
+				Err: fmt.Errorf("some error"),
+			},
+			expectHandlersKeys: []string{namespace + "/route1"},
+			expectErr:          1,
+		},
+		{
+			name: "correctly unregister route",
+			register: []routeSecret{
+				{routeName: "route1", secretName: "secret1"},
+			},
+			unregister: []routeName{"route1"},
+			expectErr:  0,
+		},
+	}
+	for _, s := range scenarios {
+		t.Run(s.name, func(t *testing.T) {
+			mgr := NewManager(nil, nil)
+			// register
+			mgr.WithSecretMonitor(&fake.SecretMonitor{}) // avoid error from AddSecretEventHandler
+			for _, rs := range s.register {
+				if err := mgr.RegisterRoute(context.TODO(), namespace, rs.routeName, rs.secretName); err != nil {
+					t.Error(err)
+				}
+			}
+
+			// unregister
+			mgr.WithSecretMonitor(&s.sm)
+			gotErr := 0
+			for _, routeName := range s.unregister {
+				if err := mgr.UnregisterRoute(namespace, string(routeName)); err != nil {
+					t.Log(err)
+					gotErr += 1
+				}
+			}
+			if gotErr != s.expectErr {
+				t.Fatalf("expected %d errors, got %d errors", s.expectErr, gotErr)
+			}
+			if len(s.expectHandlersKeys) != len(mgr.registeredHandlers) {
+				t.Fatalf("expected %d keys: %v, got %d keys: %v", len(s.expectHandlersKeys), s.expectHandlersKeys, len(mgr.registeredHandlers), mgr.registeredHandlers)
+			}
+			for _, key := range s.expectHandlersKeys {
+				if _, exists := mgr.registeredHandlers[key]; !exists {
+					t.Errorf("%s key should exist", key)
+				}
+			}
+		})
+	}
+}
+
+func TestGetSecret(t *testing.T) {
+	var (
+		namespace = "ns"
+		routeName = "route"
+	)
+
+	scenarios := []struct {
+		name      string
+		register  []routeSecret
+		sm        fake.SecretMonitor
+		expectErr bool
+	}{
+		{
+			name:      "get secret without register",
+			register:  []routeSecret{},
+			expectErr: true,
+		},
+		{
+			name:     "error from secret monitor while calling GetSecret",
+			register: []routeSecret{{routeName, "secretName"}},
+			sm: fake.SecretMonitor{
+				Err: fmt.Errorf("some error"),
+			},
+			expectErr: true,
+		},
+		{
+			name:     "successfully got secret",
+			register: []routeSecret{{routeName, "secretName"}},
+			sm: fake.SecretMonitor{
+				Err: nil,
+				Secret: &corev1.Secret{
+					Type: corev1.SecretTypeOpaque,
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "secretName",
+						Namespace: namespace,
+					},
+				},
+			},
+			expectErr: false,
+		},
+	}
+	for _, s := range scenarios {
+		t.Run(s.name, func(t *testing.T) {
+			mgr := NewManager(nil, nil)
+			// register
+			mgr.WithSecretMonitor(&fake.SecretMonitor{}) // avoid error from AddSecretEventHandler
+			for _, rs := range s.register {
+				if err := mgr.RegisterRoute(context.TODO(), namespace, rs.routeName, rs.secretName); err != nil {
+					t.Error(err)
```

Comment:
Same remark about using `Error` as opposed to `Fatal`.

Reply:
Done